incubator/hnc/api/v1alpha1/hnc_config_conversion.go

```diff
@@ -39,8 +41,21 @@ func (src *HNCConfiguration) ConvertTo(dstRaw conversion.Hub) error {
 	dstSpecTypes := []v1a2.TypeSynchronizationSpec{}
 	for _, st := range srcSpecTypes {
 		dt := v1a2.TypeSynchronizationSpec{}
-		dt.APIVersion = st.APIVersion
-		dt.Kind = st.Kind
+		// Hack the group from APIVersion by removing the version, e.g.
+		// 1) "rbac.authorization.k8s.io/v1" => "rbac.authorization.k8s.io";
+		// 2) "v1" => "" (for core type).
+		gv := strings.Split(st.APIVersion, "/")
+		if len(gv) == 2 {
+			dt.Group = gv[0]
+		}
+		// Hack the resource from Kind by using the lower case and plural form, e.g.
```

Comment:
Agreed it's a good concern. The only thing is that if we don't use hack, we would have to use a client in the conversion webhook here to look up the right resource name for the kind from apiserver. I didn't spend too much time on exploring that option though. Probably for now I will do what you suggested, filing an issue and circle back?

Reply:
So I'm in favour of leaving the hack as-is, because the cost of getting it wrong is very low - you just need to review the HNCConfiguration after installing v0.6 and fix any mistakes. If it works for all the builtin policy types, I feel like that's a win. None of this code will survive to v0.7 so I'm not too excited about spending a lot of additional time on it.

IIRC Jordan's first recommendation was not to do _any_ automatic conversion at all! So I feel this is already kind of above and beyond what's expected of a project at this level of maturity.

Thoughts?

Reply:
sgtm